user_secret.c

```diff
@@ -0,0 +1,230 @@
+#include <ctype.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+#include <syslog.h>
+#include <stdlib.h>
+
+#include "regex_helper.h"
+#include "trace.h"
+#include "user_secret.h"
+
+// use mock functions when build for UT
+#if defined (UNIT_TEST)
+void *mock_malloc(size_t size);

```

Comment:
>mock_malloc [](http://example.com/codeflow?start=6&length=11)

Will you gmock help you write mock functions?

Reply:
gmock seems only support c++ project.
the cunit example suggest using macro to mock functions like this.